### PR TITLE
Added parameterization to displayUtils.test.ts and prompts.test.ts

### PR DESCRIPTION
**Pull request by IamRiddhi**
## TLDR
Refactored test suites in `displayUtils.test.ts` and `prompts.test.ts` to use parameterized tests (`it.each`), reducing code duplication and improving readability.

## What’s Changed
- Applied parameterization (`it.each`) to repetitive test cases in `packages/cli/src/ui/utils/displayUtils.test.ts` and `packages/core/src/core/prompts.test.ts`.
- Consolidated similar tests into single blocks with shared logic and a table of test scenarios.
- Removed redundant `it` blocks and replaced them with data-driven tests.

## Why?
- **Readability:** Parameterized tests make it clear which cases are being tested, and new cases can be added easily.
- **Maintainability:** Adding or modifying edge cases only requires changing test data, not duplicating logic.
- **Coverage:** Ensures all relevant scenarios are tested in a concise format.

## Files Refactored
- `packages/cli/src/ui/utils/displayUtils.test.ts`
- `packages/core/src/core/prompts.test.ts`

## Test Plan
- All existing and new tests pass with `npm run test`.
- No change to test coverage or assertions—behavior is preserved.
- Manual review confirms tests are equivalent and easier to read.

### CODE REVIEW

File: packages/cli/src/ui/utils/displayUtils.test.ts [hunk 1]
```diff
@@ -19,54 +19,93 @@ import { Colors } from '../colors.js';
...
-        red: 20,
-      };
+      const thresholds = { green: 80, yellow: 50, red: 20 };
 
-      it('should return green for values >= green threshold', () => {
-        expect(getStatusColor(90, thresholds)).toBe(Colors.AccentGreen);
-        expect(getStatusColor(80, thresholds)).toBe(Colors.AccentGreen);
-      });
+      it.each([
+        {
```
Comment (jacob314) [inline packages/cli/src/ui/utils/displayUtils.test.ts:25]: why are there so many thresholds now?
